modules/indexing/src/test/java/org/apache/ignite/sqltests/SqlByteArrayTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.sqltests;
+
+import java.util.Arrays;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.cache.query.SqlFieldsQuery;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.internal.binary.AbstractBinaryArraysTest;
+import org.apache.ignite.internal.util.typedef.F;
+import org.junit.Test;
+
+/** Test to check CRUD with array keys and values. */
+public class SqlByteArrayTest extends AbstractBinaryArraysTest {
+    /** */
+    private static Ignite server;
+
+    /** */
+    private static Ignite client;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        server = startGrid(0);
+        client = startClientGrid(1);
+    }
+
+    /** */
+    @Test
+    public void testSQLByteArrayValue() {
```

Comment:
This test seems to work in master with the exception of byte array comparison `compareArrays(Object, Object) / BinarySerializedFieldComparator.compareArrays(Object,Object)`. If I'm correct, byte[]-value is successfully inserted/updated without the main fixes in the ticked. Do we just miss this test?

Reply:
This test is actually from IGNITE-5464.